src/ray/gcs/redis_context.cc

```diff
@@ -384,6 +388,63 @@ Status RedisContext::PingPort(const std::string &address, int port) {
       address, port, redisConnect, static_cast<redisContext **>(nullptr));
 }
 
+void ValidateRedisDB(RedisContext &context) {
+  auto reply = context.RunArgvSync(std::vector<std::string>{"INFO", "CLUSTER"});
+  // cluster_state:ok
+  // cluster_slots_assigned:16384
+  // cluster_slots_ok:16384
+  // cluster_slots_pfail:0
+  // cluster_size:1
+  RAY_CHECK(reply && !reply->IsNil()) << "Failed to get Redis cluster info";
+  auto cluster_info = reply->ReadAsString();
+
+  std::vector<std::string> parts = absl::StrSplit(cluster_info, "\r\n");
+  bool cluster_mode = false;
+  int cluster_size = 0;
+
+  // Check the cluster status first
+  for (auto &part : parts) {
+    if (part.empty() || part[0] == '#') {
```

Comment:
Is there any way to test this method with real redis? Is it in CI test?

Reply:
We have redis e2e test for this one. Do you mean testing the parsing part only ?